torch/_meta_registrations.py

```diff
@@ -309,6 +309,32 @@ def meta_unsqueeze_(self, dim):
     return self
 
 
+@register_meta(aten.index_reduce.default)
+def meta_index_reduce(
+    self: Tensor,
+    dim: int,
+    index: Tensor,
+    source: torch.Tensor,
+    reduce: str,
+    *,
+    include_self: bool = True,
+) -> Tensor:
+    return torch.empty_like(self).contiguous()
```

Comment:
```suggestion
    return torch.empty_like(self, memory_format=torch.contiguous_format)
```

But the kernel actually always return a contiguous Tensor vs preserving the input one?

Reply:
yes, at least I have one example where a transposed input produces a contiguous output

```python
import torch
import torch._dynamo
from torch.testing import make_tensor
from torch import tensor

def fn(x, y, t):
    return x.index_reduce(dim=0, index=y, source=t, reduce="prod", include_self=True)  # .sum()

x = torch.empty(3, 5).fill_(2).cuda().T
t = torch.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9], [10, 11, 12]], dtype=torch.float, device='cuda')
index = torch.tensor([0, 4, 2, 0]).cuda()
args = [x, index, t]

out = fn(*args)
print(x.is_contiguous(), out.is_contiguous())
# False, True
```